### PR TITLE
Update default-versions.tf

### PR DESCRIPTION
**Pull request by JohnbelMDev**
Regularly audit and update provider versions for security patches. Use a secure backend with encryption for the Terraform state. Validate input variables and restrict provider permissions. Avoid exposing unnecessary metadata (like release notes) unless necessary for internal documentation.

---
**Checklist**

I applicable, I acknowledge that I have:
- [ ] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [ ] Ran `terraform fmt` on all modified files
- [ ] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [ ] Made sure all relevant tests pass

### CODE REVIEW

**Comment by ludoo** [comment]
I think we're fine with our current approach. @juliocc @wiktorn WDYT?

**Comment by wiktorn** [comment]
@JohnbelMDev I think that better approach to achieve what you're looking for is to use [.terraform.lock.hcl](https://developer.hashicorp.com/terraform/language/files/dependency-lock).

`versions.tf` file say about the compatibility - so what we do expect to work with the given code. While users may mix modules from different sources in one code they need to rely on more relaxed versioning to allow module upgrades. Otherwise all modules that they use has to agree on using exactly the same version of the provider which makes then upgrades close to impossible.

**Comment by juliocc** [comment]
Actually I think this is a good idea. For one, it would remove the need for [pinning versions before running tests](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/.github/actions/fabric-tests/action.yml#L71). It is definitely more conservative than our current approach but I think it would prevent some issues we've seen in the past.

I'd do it only for the provider, leaving the Terraform version as we currently have it.

**Comment by ludoo** [comment]
Well, decide between you two @juliocc and @wiktorn  :)

File: default-versions.tf [hunk 1]
```diff
@@ -12,18 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Fabric release: v36.0.1
```
Comment (ludoo) [inline default-versions.tf:15]: Please do not remove this.

**Comment by juliocc** [comment]
After a bit more discussion with @wiktorn we've decided not to go with this. Thanks

**Comment by JohnbelMDev** [comment]
Thank you for your thoughtful response and guidance of insights for
providing relevant details . I will take into consideration of the
conservative to be more modern approach. Thank you indicating the approach
wasn't modern to modernize code standard of google that still getting
familiar to me.  Will be more global thinker for flexible for
situations like this of current. Terrafor.lock.hcl are usually there for
versions.


Best regards,


-Johnbel Mahautiere

**Pull request closed without merging**